### PR TITLE
Compile fixes

### PR DESCRIPTION
**Pull request by JohnLangford**
Fix the last compile warnings.

### CODE REVIEW

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -1053,7 +1053,7 @@ void parse_output_preds(options_i& options, vw& all)
       if (options.was_supplied("binary"))
         all.trace_message << "Warning: --raw_predictions has no defined value when --binary specified, expect no output" << endl;
     }
-    if (raw_predictions.c_str() == "stdout")
+    if (strcmp(raw_predictions.c_str(),"stdout")==0)
```
Comment (jackgerrits) [inline vowpalwabbit/parse_args.cc]: Why not just compare the string object to the temporary?

raw_predictions == “stdout”

Comment (JohnLangford) [reply vowpalwabbit/parse_args.cc]: That sounds plausibly better.

Comment (JohnLangford) [reply vowpalwabbit/parse_args.cc]: (Send a pull request?)

**Comment by jackgerrits** [comment]
Will do

**Pull request merged**